bpf/bpf_lxc.c

```diff
@@ -328,7 +330,7 @@ static inline int ipv6_l3_from_lxc(struct __sk_buff *skb,
 	}
 
 	/* The packet goes to a peer not managed by this agent instance */
-#ifdef ENCAP_IFINDEX
+#if defined(ENCAP_IFINDEX) && !defined(DATAPATH_IPVLAN)
```

Comment:
We can use `option.Config.Tunnel = option.TunnelDisabled` in the agent when we run in IPVLAN mode and it will result in `ENCAP_IFINDEX` not being set.

Reply:
Agree, I was planning to remove this here anyway like I did in bpf_netdev because tunnel mode needs to be disabled in order to select ipvlan data path mode, so this is not needed.